packages/core/src/modules/payment/payment-service.ts

```diff
@@ -199,8 +199,9 @@ export class PaymentServiceImpl implements PaymentService {
   }
 
   async refundPayment(payment: Payment, product: Product): Promise<void> {
-    const paymentProvider = product.paymentProviders.find((p) => p.paymentProviderId === payment.paymentProviderId)
-      ?.paymentProvider
+    const paymentProvider = product.paymentProviders.find(
```

Comment:
This seems unrelated. If it was from a manual change please avoid doing so in the future, otherwise it might be from a dependency-update

Reply:
undone